src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/CollectionModelBinder.cs

```diff
@@ -93,13 +90,20 @@ public class CollectionModelBinder<TElement> : ICollectionModelBinder
             }
 
             bindingContext.Result = ModelBindingResult.Success(bindingContext.ModelName, model);
-            return;
         }
 
         /// <inheritdoc />
         public virtual bool CanCreateInstance(Type targetType)
         {
-            return CreateEmptyCollection(targetType) != null;
+            if (targetType.IsAssignableFrom(typeof(List<TElement>)))
```

Comment:
Should result in slightly better performance because it's no longer creating an instance to check if it can create an instance. Touches on #4181 as well.